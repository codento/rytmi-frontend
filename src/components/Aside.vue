<template>
  <aside class="aside-menu" />
</template>

<script>
export default {
  name: 'c-aside'
}
</script>
