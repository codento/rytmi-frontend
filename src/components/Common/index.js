import ProjectRow from './ProjectRow.vue'
import SkillRow from './SkillRow.vue'
import SkillExplanations from './SkillExplanations.vue'
import UtilizationChart from './UtilizationChart.vue'
import CollapsableItem from './CollapsableItem.vue'
import EducationCard from './EducationCard.vue'
import ProjectForm from './ProjectForm.vue'
import ProjectProfileForm from './ProjectProfileForm.vue'

export {
  ProjectRow,
  SkillRow,
  SkillExplanations,
  UtilizationChart,
  CollapsableItem,
  EducationCard,
  ProjectForm,
  ProjectProfileForm
}
