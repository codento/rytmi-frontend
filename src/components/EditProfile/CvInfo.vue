<template>
  <div class="my-4 animated fadeIn">
    <h1>Information for CV</h1>
    <hr>
    <CvInfoEditIntroduction :profile="profile" />
    <hr>
    <CvInfoEditEducation :profile="profile" />
  </div>
</template>

<script>
import CvInfoEditIntroduction from './CvInfoEditIntroduction'
import CvInfoEditEducation from './CvInfoEditEducation'

export default {
  name: 'CvInfo',
  components: {
    CvInfoEditIntroduction,
    CvInfoEditEducation
  },
  props: {
    profile: Object
  },
  data () {
    return {}
  },
  computed: {},
  methods: {}
}
</script>
