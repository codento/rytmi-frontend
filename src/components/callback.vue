<template>
  
</template>

<script>
import {mapActions} from 'vuex'

export default {
  created () {
    this.$nextTick(() => {
      this.fetchProfiles()
      this.fetchSkills()
      this.fetchProfileSkills()
      this.$router.push(this.$route.query.redirect || '/')
    })
  },
  methods: {
    ...mapActions([
      'fetchProfiles',
      'fetchSkills',
      'fetchProfileSkills'
    ])
  }
}
</script>