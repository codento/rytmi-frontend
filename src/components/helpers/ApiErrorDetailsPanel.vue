<template>
  <ul>
    <li
      v-for="detail in errorDetails"
      :key="detail"
      class="error-detail-row"
    >
      {{ parseErrorDetail(detail) }}
    </li>
  </ul>
</template>
<script>
export default {
  name: 'ApiErrorDetailsPanel',
  props: {
    errorDetails: Array
  },
  methods: {
    parseErrorDetail (detail) {
      return detail.replace('.', ' ').replace('null', 'empty')
    }
  }
}
</script>
