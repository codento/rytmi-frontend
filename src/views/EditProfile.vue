<template>
  <div class="animated fadeIn profile-editor col-sm-12 col-md-10">
    <h1>Edit Profile</h1>
    <hr />
    <h1>Basic Details</h1>
    <hr />
    <template v-if="profile">
      <ProfileForm :profile="profile" />
      <EditLinks :profile="profile" />
      <edit-skills :profileId="profile.id" />
      <edit-projects :profileId="profile.id" />
    </template>
  </div>
</template>

<script>
import { mapGetters } from 'vuex'
import {
  EditLinks,
  EditSkills,
  EditProjects,
  ProfileForm
} from '../components/EditProfile'
import store from '../store'

export default {
  name: 'EditProfile',
  props: [
    'profileId'
  ],
  components: {
    ProfileForm,
    EditLinks,
    EditSkills,
    EditProjects
  },
  computed: {
    ...mapGetters([
      'profileById'
    ]),
    profile () {
      return this.profileById(this.profileId)
    }
  },
  mounted () {
    document.title = 'Rytmi - Edit Profile'
  },
  beforeRouteEnter (to, from, next) {
    store.dispatch('fetchProfileProjects', to.params.profileId)
    next()
  }
}
</script>

<style scoped>
.profile-editor {
  margin: 0 auto;
}
</style>
