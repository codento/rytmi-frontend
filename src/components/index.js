import Aside from './appStructures/Aside.vue'
import Breadcrumb from './appStructures/Breadcrumb.vue'
import Callout from './helpers/Callout.vue'
import Footer from './appStructures/Footer.vue'
import Header from './appStructures/Header.vue'
import Sidebar from './appStructures/Sidebar/Sidebar.vue'

export {
  Aside,
  Breadcrumb,
  Callout,
  Footer,
  Header,
  Sidebar
}
