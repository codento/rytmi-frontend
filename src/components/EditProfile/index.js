import EditLinks from './EditLinks.vue'
import EditSkills from './EditSkills.vue'
import ProfileForm from './ProfileForm.vue'
import EditProjects from './EditProjects.vue'
import WorkHistoryEntryEdit from './WorkHistoryEntryEdit.vue'
import WorkHistoryEntryList from './WorkHistoryEntryList.vue'
import CvInfo from './CvInfo.vue'
import CvInfoEditIntroduction from './CvInfoEditIntroduction.vue'
import CvInfoEditEducation from './CvInfoEditEducation.vue'
import CvInfoEditEducationForm from './CvInfoEditEducationForm.vue'

export {
  EditLinks,
  EditSkills,
  ProfileForm,
  EditProjects,
  WorkHistoryEntryList,
  CvInfo,
  CvInfoEditIntroduction,
  CvInfoEditEducation,
  CvInfoEditEducationForm,
  WorkHistoryEntryEdit
}
