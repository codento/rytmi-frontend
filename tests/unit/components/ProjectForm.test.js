import { merge, cloneDeep } from 'lodash'
import flushPromises from 'flush-promises'
import { ProfileForm } from '@/components/EditProfile'
import ApiErrorDetailsPanel from '@/components/helpers/ApiErrorDetailsPanel'

import { createWrapper } from './setup/setup'

const mockEmployeeRoles = [
  { id: 1, title: 'somethinger' },
  { id: 2, title: 'dunno lol' }
]

const defaultStoreConfig = {
  actions: {
    updateProfile: jest.fn(() => [])
  },
  getters: {
    employeeRoles: jest.fn(() => mockEmployeeRoles)
  }
}

const mockProfile = {
  id: 2,
  firstName: 'Foo',
  lastName: 'Bar',
  role: 'Employee role',
  birthYear: 1984,
  title: 'Title',
  phone: 1234,
  information: {
    fi: 'kuvaus1',
    en: 'desc1'
  },
  education: [
    { fi: { school: 'Yliopisto', degree: 'Joku tutkinto', major: null, minor: null },
      en: { school: 'University', degree: 'Some degree', major: 'My major', minor: 'My minor' },
      startYear: 2011,
      endYear: 2015
    }
  ],
  email: 'foo.bar@barfoo.com',
  employeeRoles: [1],
  links: ['http://christy.net', 'http://holly.biz', 'http://www.linkedin.com/username']
}

const defaultMountingOptions = {
  propsData: {
    profile: mockProfile
  },
  mocks: {
    $toasted: {
      global: {
        rytmi_success: jest.fn()
      }
    }
  }
}

describe('ProfileForm.vue', () => {
  it('should should show profile details correctly', () => {
    const wrapper = createWrapper(ProfileForm, defaultStoreConfig, defaultMountingOptions)
    expect(wrapper.vm.editedProfile).toEqual(mockProfile)
    const inputWrappers = wrapper.findAll('input')
    expect(inputWrappers.at(0).vm.value).toBe(mockProfile.firstName)
    expect(inputWrappers.at(1).vm.value).toBe(mockProfile.lastName)
    expect(inputWrappers.at(3).vm.value).toBe(mockProfile.birthYear)
    expect(inputWrappers.at(4).vm.value).toBe(mockProfile.email)
    expect(inputWrappers.at(5).vm.value).toBe(mockProfile.phone)
  })

  it('should submit entered details when submit is clicked', async () => {
    expect.assertions(3)
    const editedProfile = cloneDeep(mockProfile)
    editedProfile.firstName = 'Bar'

    const mocks = {
      $router: {
        push: jest.fn()
      }
    }
    const actions = {
      updateProfile: jest.fn((mapActionsStuff, profile, undef) => Promise.resolve(profile))
    }
    const overrideStoreConfig = merge({}, defaultStoreConfig, { actions })
    const overrideMountingOptions = merge({}, defaultMountingOptions, { mocks })
    const wrapper = createWrapper(ProfileForm, overrideStoreConfig, overrideMountingOptions)
    wrapper.find('#firstNameInput').setValue('Bar')
    wrapper.find('button[type="submit"]').trigger('submit')
    await flushPromises()
    expect(actions.updateProfile).toHaveBeenCalledWith(
      expect.anything(), editedProfile, undefined)
    expect(wrapper.vm.showError).toBe(false)
    expect(wrapper.vm.errorDetails).toHaveLength(0)
  })

  it('should go back to profile view when reset is clicked', () => {
    expect.assertions(3)
    const mocks = {
      $router: {
        push: jest.fn()
      }
    }
    const wrapper = createWrapper(ProfileForm, defaultStoreConfig, merge({}, defaultMountingOptions, { mocks }))
    wrapper.find('button[type="reset"]').trigger('reset')
    expect(mocks.$router.push).toHaveBeenCalledWith(`/profile/${mockProfile.id}`)
    expect(wrapper.vm.showError).toBe(false)
    expect(wrapper.vm.errorDetails).toHaveLength(0)
  })

  it('shows error when submit fails', async () => {
    expect.assertions(3)
    const mocks = {
      $router: {
        push: jest.fn()
      }
    }
    const apiError = {
      details: ['Nope not today']
    }
    const actions = {
      updateProfile: jest.fn((mapActionsStuff, profile, undef) => Promise.reject(apiError))
    }
    const editedProfile = cloneDeep(mockProfile)
    editedProfile.firstName = 'Bar'

    const overrideStoreConfig = merge({}, defaultStoreConfig, { actions })
    const overrideMountingOptions = merge({}, defaultMountingOptions, { mocks })
    const wrapper = createWrapper(ProfileForm, overrideStoreConfig, overrideMountingOptions)
    wrapper.find('button[type="submit"]').trigger('submit')
    await flushPromises()
    expect(wrapper.vm.showError).toBe(true)
    expect(wrapper.find(ApiErrorDetailsPanel).isVisible()).toBeTruthy()
    expect(wrapper.vm.errorDetails).toHaveLength(1)
  })

  it('returns correct employeeRoleList', () => {
    const wrapper = createWrapper(ProfileForm, defaultStoreConfig, defaultMountingOptions)
    wrapper.setData({ selectedEmployeeRoles: [] })
    expect(wrapper.vm.employeeRoleList).toEqual([{ 'id': 1, 'label': 'somethinger' }, { 'id': 2, 'label': 'dunno lol' }])
  })
})
