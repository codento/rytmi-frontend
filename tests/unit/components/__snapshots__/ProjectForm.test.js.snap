// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectForm.test.js Template is correct 1`] = `
<div>
  <h3 style="text-align: center;">
    Add a new project
    <span><i class="fa fa-chevron-down"></i></span></h3>
  <b-form id="project_form" class="animated fadeIn" style="display: none;">
    <b-row>
      <b-col><small>Project code</small>
        <b-input placeholder="Project code" required="" type="number" min="0" max="99999"></b-input>
      </b-col>
    </b-row>
    <b-row>
      <b-col sm="6"><small>Project name (Finnish)</small>
        <b-input placeholder="Project name (fi)" required="" type="text"></b-input>
      </b-col>
      <b-col sm="6"><small>Project name (English)</small>
        <b-input placeholder="Project name (en)" required="" type="text"></b-input>
      </b-col>
    </b-row>
    <b-row>
      <b-col><small>Start date</small>
        <div><input id="project-start-date-datepicker" type="text" autocomplete="off" class="form-control"></div>
      </b-col>
    </b-row>
    <b-row>
      <b-col><small>End date</small>
        <div><input id="project-end-date-datepicker" type="text" autocomplete="off" class="form-control"></div>
      </b-col>
    </b-row>
    <b-row>
      <b-col sm="6"><small>Description (Finnish)</small>
        <b-textarea placeholder="Project description (fi)" type="text" rows="5" class="form-control"></b-textarea>
      </b-col>
      <b-col sm="6"><small>Description (English)</small>
        <b-textarea placeholder="Project description (en)" type="text" rows="5" class="form-control"></b-textarea>
      </b-col>
    </b-row>
    <b-row>
      <b-col>
        <b-button type="submit" primary="" class="form-control">
          Submit
        </b-button>
      </b-col>
    </b-row>
  </b-form>
  <!---->
</div>
`;
